src/geom/util/PointExtracter.cpp

```diff
@@ -28,6 +28,10 @@ namespace util { // geos.geom.util
 void
 PointExtracter::getPoints(const Geometry& geom, Point::ConstVect& ret)
 {
+    if (!geom.hasDimension(Dimension::P)) {
```

Comment:
What if geom is a `GeometryCollection` with points as well as higher-dimension elements?